README.md

```diff
@@ -1,134 +1,106 @@
+1. Install dependencies:
+```bash
+pnpm install
+```
 
-## Build
+2. Copy `.env.example` to `.env` and configure:
+```bash
+cp .env.example .env
+```
 
-### Compiling your contracts
+3. (optional) Set deployer credentials in `.env`:
+```
+MNEMONIC="your mnemonic here"
+# or
+PRIVATE_KEY="0xabc...def"
+```
 
-<!-- TODO: consider moving this section to Appendix, since for Hardhat, the deploy task wil auto-run compile -->
+4. (optional) Fund the deployer address with native tokens on target chains
 
-This project supports both `hardhat` and `forge` compilation. By default, the `compile` command will execute both:
+## Compilation
 
+Compile contracts using both Hardhat and Forge:
 ```bash
 pnpm compile
 ```
 
-If you prefer one over the other, you can use the tooling-specific commands:
-
+Or compile individually:
 ```bash
 pnpm compile:forge
 pnpm compile:hardhat
 ```
 
-## Deploy
+**NOTE:** you may see a warning about `PRIVATE_KEY` environment variable not being read if you skipped step (3) above. This is normal.
+
+## Deployment
 
-First, deploy the inner token to (only) **Optimism Sepolia**.
+### 1. Configure Token Address
+
+**NOTE:** In general this step can be skipped because the change has already been applied to the repo. This information is left here in case you are deploying a test/forked token.
 
 In `hardhat.config.ts`, set the token address for the OFTAdapter on the origin chain. `<INNER_TOKEN_ADDRESS>` should be replaced with the native address of your token on its main network (which, in the case of CoW, is [`0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB`](https://docs.cow.fi/governance/token) on mainnet):
 
 ```typescript
-// Replace `0x0` with the address of the ERC20 token you want to adapt to the OFT functionality.
 oftAdapter: {
     tokenAddress: '<INNER_TOKEN_ADDRESS>',
 }
 ```
 
-Deploy an OFTAdapter to Optimism Sepolia:
+### 2. (optional) Configure Multisig Owner
 
-```bash
-pnpm hardhat lz:deploy --tags MyOFTAdapter --networks optimism-testnet
+Set up a Safe multisig at [https://app.safe.global](https://app.safe.global) and configure it in `hardhat.config.ts`:
+
+```typescript
+owner: {
+    default: '<SAFE_ADDRESS>', // CoW DAO multisig
+},
 ```
 
-Deploy the OFT to Arbitrum Sepolia:
+### 3. Deploy Contracts
 
+Deploy the OFTAdapter on the origin chain (which, in CoW's case, is mainnet):
 ```bash
-pnpm hardhat lz:deploy --tags MyOFT --networks arbitrum-testnet
+pnpm hardhat lz:deploy --tags CowOftAdapter --networks mainnet
 ```
 
-### Verifying Deployed Contracts Manually
+Deploy OFT contracts on destination chains. You can find a list of chains to deploy on, or add your own, in `hardhat.config.ts`:
+```bash
+pnpm hardhat lz:deploy --tags CowOft --networks <destination-network>
+```
+
+#### Manually Verifying the Deployed Contracts
```

Comment:
The verification process here is misleading since one could deploy a contract that passes this check but that at the same time sets the contract state to something unexpected (for example, preallocating some tokens to some address). The same check should work if we used the (deployment) bytecode and compare it with the create2 input of the transaction (which is unfortunately harder to do programmatically).